src/som/interpreter/actors/Actor.java

```diff
@@ -195,44 +220,74 @@ public void run() {
 
     private void processCurrentMessages(final ActorProcessingThread currentThread, final WebDebugger dbg) {
       if (VmSettings.ACTOR_TRACING) {
-        current.setExecutionStart(System.nanoTime());
-        current.setBaseMessageId(currentThread.generateMessageBaseId(current.size()));
-        currentThread.currentMessageId = current.getBasemessageId();
+        baseMessageId = currentThread.generateMessageBaseId(size);
+        currentThread.currentMessageId = baseMessageId;
       }
-      for (EventualMessage msg : current) {
-        actor.logMessageBeingExecuted(msg);
-        currentThread.currentMessage = msg;
 
-        if (VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
-          if (msg.isBreakpoint()) {
+      if (size > 0) {
+        actor.logMessageBeingExecuted(current);
+        currentThread.currentMessage = current;
+
+        if (VmSettings.TRUFFLE_DEBUGGER_ENABLED && current.isBreakpoint()) {
             dbg.prepareSteppingUntilNextRootNode();
-          }
         }
-        if (VmSettings.MESSAGE_TIMESTAMPS) {
-          current.addMessageExecutionStart();
-        }
-        msg.execute();
+
+        current.execute();
         if (VmSettings.ACTOR_TRACING) {
           currentThread.currentMessageId += 1;
         }
+
+        int i = 0;
+        if (size > 1) {
+          for (EventualMessage msg : moreCurrent) {
+            actor.logMessageBeingExecuted(msg);
```

Comment:
this block looks identical to the handling of the first message. could this be a helper method?